### PR TITLE
Analyzer: Fix assertion in HashJoin with duplicate columns

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 83007c59daf83d1962981701ee9cf0e64ff19e19 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58652/83007c59daf83d1962981701ee9cf0e64ff19e19/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58652/83007c59daf83d1962981701ee9cf0e64ff19e19/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Interpreters/HashJoin.cpp [hunk 1]
```diff
@@ -2009,12 +2002,14 @@ struct AdderNonJoined
 /// Based on:
 ///   - map offsetInternal saved in used_flags for single disjuncts
 ///   - flags in BlockWithFlags for multiple disjuncts
-template <bool multiple_disjuncts>
```
Comment (vdimir) [inline src/Interpreters/HashJoin.cpp:2012]: This template seemed to be useless

Comment (novikd) [reply src/Interpreters/HashJoin.cpp:2012]: The only possible difference -- performance. Better to check Performance Tests results.

File: src/Interpreters/HashJoin.cpp [hunk 2]
```diff
@@ -2213,14 +2209,14 @@ IBlocksStreamPtr HashJoin::getNonJoinedBlocks(const Block & left_sample_block,
     {
         /// ... calculate `left_columns_count` ...
         size_t left_columns_count = left_sample_block.columns();
-        auto non_joined = std::make_unique<NotJoinedHash<true>>(*this, max_block_size);
+        auto non_joined = std::make_unique<NotJoinedHash>(*this, max_block_size, multiple_disjuncts);
         return std::make_unique<NotJoinedBlocks>(std::move(non_joined), result_sample_block, left_columns_count, *table_join);
     }
     else
```
Comment (novikd) [inline src/Interpreters/HashJoin.cpp:2215]: If you remove the template parameter there's no need for this if-statement.

**Review by novikd — APPROVED** [review]
LGTM

**Comment by vdimir** [comment]
> Stateless tests (debug) [5/5] — fail: 1, passed: 1191, skipped: 9 [Details](https://s3.amazonaws.com/clickhouse-test-reports/58652/61f2ab3ec5420a3c036eb53ec22e26f391392a6e/stateless_tests__debug__[5_5].html)

`01961_roaring_memory_tracking` is slow with `max_bytes_before_external_group_by = 1` https://github.com/ClickHouse/ClickHouse/pull/58863

> Stateless tests (tsan, s3 storage) [5/5] — Tests are not finished, fail: 1, passed: 1026, skipped: 13 [Details](https://s3.amazonaws.com/clickhouse-test-reports/58652/61f2ab3ec5420a3c036eb53ec22e26f391392a6e/stateless_tests__tsan__s3_storage__[5_5].html)

https://github.com/ClickHouse/ClickHouse/issues/58437

**Pull request merged**